3_Repositories.Tests/DataSources/ProjectDataSourceAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿using _3_Repositories.Tests.DataGenerators;
+using _3_Repositories.Tests.DataGenerators.Base;
+using Models;
+using NUnit.Framework.Interfaces;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace _3_Repositories.Tests.DataSources
+{
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+    public class ProjectDataSourceAttribute : Attribute, IParameterDataSource
+    {
+        private readonly IFakeDataGenerator<Project> _fakeDataGenerator;
```

Comment:
The coding guidelines for this project do not allow an underscore prefix.
```suggestion
        private readonly IFakeDataGenerator<Project> fakeDataGenerator;
```